dist/activate

```diff
@@ -0,0 +1,171 @@
+  if [[ ! -x "${KCTF_BIN}/yq" ]]; then
+    if [[ -e "${KCTF_BIN}/yq" ]]; then
+      rm "${KCTF_BIN}/yq" >/dev/null
+    fi
+    wget "https://github.com/mikefarah/yq/releases/download/v4.2.0/yq_linux_amd64" -O "${KCTF_BIN}/yq" --quiet || return 1
+    sha256sum --status -c <(echo "5d44bd64e264e9029c5f06bcd960ba162d7ed7ddd1781f02a28d62f50577b632  ${KCTF_BIN}/yq") || return 1
+    chmod u+x "${KCTF_BIN}/yq"
+  fi
+}
+
+function _kctf_export_variables {
+  export KCTF_CTF_DIR
+  export KCTF_CTF_NAME
+  export KCTF_BIN
+  export CLOUDSDK_ACTIVE_CONFIG_NAME="kctf-${KCTF_SESSION}"
+  export KUBECONFIG="${KCTF_CONFIG_DIR}/kube-${KCTF_SESSION}.conf"
+}
+
+function _kctf_cleanup {
+  if command -v gcloud >/dev/null 2>&1; then
+    unset CLOUDSDK_ACTIVE_CONFIG_NAME
+    # regenerate this name in case the user changed the variable
+    GCLOUD_CONFIG_NAME="kctf-${KCTF_SESSION}"
+    if gcloud config configurations describe "${GCLOUD_CONFIG_NAME}" >/dev/null 2>&1; then
+      echo "Deleting gcloud config ${GCLOUD_CONFIG_NAME}"
+      CLOUDSDK_CORE_DISABLE_PROMPTS=1 gcloud config configurations delete "${GCLOUD_CONFIG_NAME}"
+    fi
+  fi
+  # regenerate this name in case the user changed the variable
+  KUBE_CONFIG_NAME="${KCTF_CONFIG_DIR}/kube-${KCTF_SESSION}.conf"
+  if [[ -e "${KUBE_CONFIG_NAME}" ]]; then
+    rm "${KUBE_CONFIG_NAME}" >/dev/null
+  fi
+}
+
+function _kctf_usage {
+  echo -e "usage: kctf command subcommand [args]" >&2
+  echo -e "available commands:" >&2
+  echo -e "  chal:    commands around challenges (creating, deploying, etc.)" >&2
+  echo -e "  cluster: commands for managing the cluster " >&2
+  echo -e "  config:  create/load a cluster config" >&2
+}
+
+# Implemented as a function so that we can set environment variables where needed
+function kctf {
+  if [[ $# -lt 1 ]]; then
+    echo "missing required argument" >&2
+    _kctf_usage
+    return 1
+  fi
+  case "$1" in
+    -h|--help)
+      _kctf_usage
+      return 0
+      ;;
+    chal)
+      shift
+      "${KCTF_CTF_DIR}/kctf/bin/kctf-challenge" $@
+      return
+      ;;
+    cluster)
+      shift
+      "${KCTF_CTF_DIR}/kctf/bin/kctf-cluster" $@
+      return
+      ;;
+    config)
+      shift
+      if [[ "$1" == "create" ]] || [[ "$1" == "load" ]]; then
+        CONFIG_NAME=$("${KCTF_CTF_DIR}/kctf/bin/kctf-config" $@)
+        if [[ $? -ne 0 ]]; then
+          return 1
+        fi
+        source "${KCTF_CTF_DIR}/kctf/config/${CONFIG_NAME}"
+        export PROJECT
+        export ZONE
+        export REGISTRY
+        export CLUSTER_NAME
+        export DOMAIN_NAME
+        KCTF_CONFIG="${CONFIG_NAME}"
+      else
+        "${KCTF_CTF_DIR}/kctf/bin/kctf-config" $@
+      fi
+      return
+      ;;
+    *)
+      _kctf_usage
+      return 1
+      ;;
+  esac
+}
+
+function _kctf_enable_completion {
+  source "${KCTF_BIN}/kctf-completion"
+}
+
+function _kctf_error_cleanup {
+  unset -f _kctf_download_dependencies
+  #unset -f _kctf_cleanup
```

Comment:
?

Reply:
I think I didn't figure out how to remove the trap, so this function will get called on exit

Reply:
I meant, the `#` commented line :)

Reply:
I know, I'm talking about _kctf_cleanup. Added a comment